projects/components/src/load-async/loader/loader.component.ts

```diff
@@ -1,15 +1,37 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
-import { IconType } from '@hypertrace/assets-library';
-import { IconSize } from '../../icon/icon-size';
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { ImagesAssetPath, LoaderType } from '@hypertrace/assets-library';
+import { assertUnreachable } from '@hypertrace/common';
 
 @Component({
   selector: 'ht-loader',
   styleUrls: ['./loader.component.scss'],
   template: `
     <div class="ht-loader">
-      <ht-icon icon="${IconType.Loading}" size="${IconSize.Large}"></ht-icon>
+      <img [ngClass]="[this.type]" [src]="this.imagePath" />
     </div>
   `,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class LoaderComponent {}
+export class LoaderComponent implements OnChanges {
+  @Input()
+  public type!: LoaderType;
+
+  public imagePath: ImagesAssetPath = ImagesAssetPath.LoaderExpandableRow;
+
+  public ngOnChanges(): void {
+    this.imagePath = this.getImagePathFromType();
+  }
+
+  private getImagePathFromType(): ImagesAssetPath {
+    switch (this.type) {
+      case LoaderType.ExpandableRow:
+        return ImagesAssetPath.LoaderExpandableRow;
+      case LoaderType.Page:
+        return ImagesAssetPath.LoaderPage;
+      case LoaderType.Spinner:
+        return ImagesAssetPath.LoaderSpinner;
+      default:
+        return assertUnreachable(this.type);
```

Comment:
may be we can leave `type` as optional and in default section here, we can assign the default image path instead of throwing exception

Reply:
Actually, No. When a consumer is passing a wrong value it should throw an error, on the other hand, if the consumer is not passing a value it should show default value.